mlflow/genai/scorers/builtin_scorers.py

```diff
@@ -36,6 +41,88 @@
 from mlflow.genai.judges.base import Judge, JudgeField
 
 
+@dataclass
+class ExtractedFields:
+    inputs: Any | None = None
+    outputs: Any | None = None
+    expectations: dict[str, Any] | None = None
+
+
+def _extract_fields_from_trace(
```

Comment:
Nit - should this just be called `resolve_scorer_fields` or something? It doesn't always extract the fields from a trace.